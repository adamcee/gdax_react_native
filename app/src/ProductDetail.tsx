// ProductDetail.tsx
// Show detailed information for a GDAX Product 
import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { GDAXProduct } from './interfaces/gdax_product';

interface ComponentProps {
  product: GDAXProduct;
}

interface ComponentState {
}

export default class ProductDetail extends React.Component<ComponentProps, ComponentState> {
    constructor(props: any) {
        super(props);
    }

    componentDidMount() {
        console.log('PRODUCT DETAIL: ', this.props.product);
    }

    render() {
        const { product } = this.props;
        const productDetailNames = Object.keys(product);
        return (
            <View>
                {productDetailNames.map((detail, i) =>
                    <Text key={i}>{detail}: {product[detail]}</Text>
                )}
            </View>
        );
    }
}