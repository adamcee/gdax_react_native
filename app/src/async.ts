// async.ts
// functions for asynchronous http calls

export function getJSONResource(host: string, resourceName: string) {
    return getResource(host, resourceName)
        .then(response => response.json());
}

export function getResource(host: string, resourceName: string) {
    return fetch(`${host}/${resourceName}`)
        .then(handleErrors);
}

export function handleErrors(response: Response) {
    if(!response.ok) {
        throw Error(response.statusText);
    }
    return response;
}