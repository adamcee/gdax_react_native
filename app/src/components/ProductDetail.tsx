// ProductDetail.tsx
// Show detailed information for a GDAX Product 
import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { Card, Text } from 'react-native-elements';
import { GDAXProduct } from '../interfaces/gdax_product';

interface ComponentProps {
  product: GDAXProduct;
}

interface ComponentState {
}

export default class ProductDetail extends React.Component<ComponentProps, ComponentState> {
    constructor(props: any) {
        super(props);
    }

    render() {
        const { product } = this.props;
        const productDetailNames = Object.keys(product);
        return (
            <Card wrapperStyle={{ marginLeft: 20, marginRight: 20}} >
                {productDetailNames.map((detail, i) =>
                    <Text key={i}><Text h4>{detail}</Text>: {product[detail]}</Text>
                )}
            </Card>
        );
    }
}