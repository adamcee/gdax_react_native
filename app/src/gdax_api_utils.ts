// gdax_api_utils.ts
// helper functions for the gdax api and manipulating data it returns
import { GDAXProduct, GDAXProductMap } from './interfaces/gdax_product';


// Takes an array of products with various base currencies 
// and returns a dict grouped by base currency.
export function groupProductsByBaseCurrency(products: Array<GDAXProduct>): GDAXProductMap {
    let container: GDAXProductMap = {};

    const groupedProducts = products.reduce(groupProductsByBase, container);

    function groupProductsByBase(container: GDAXProductMap, product: GDAXProduct): any {
        const base_currency = product.base_currency;
        if (!container.hasOwnProperty(base_currency)) {
            container[base_currency] = [] as Array<GDAXProduct>;
        } else {
            container[base_currency].push(product);
        }
        return container;
    }

    return groupedProducts;
}