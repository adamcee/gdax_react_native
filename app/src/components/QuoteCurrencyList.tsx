// QuoteCurrencyList.tsx
import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button } from 'react-native-elements';
import { GDAXProduct } from '../interfaces/gdax_product';
import { toggleElInUniqueStringArray } from '../utils';
import ProductDetail from './ProductDetail';

interface ComponentProps {
    qcProds: ReadonlyArray<GDAXProduct>;
}

interface ComponentState {
  selectedProductIds: Array<string>;
}

export default class QuoteCurrencyList extends React.Component<ComponentProps, ComponentState> {
  constructor(props: any) {
    super(props);
    this.state = {
      selectedProductIds: [],
    };
    this.toggleProductInfo = this.toggleProductInfo.bind(this);
  }

  // toggle if info for a product is displayed or not.
  toggleProductInfo(productId: string) {
    const { selectedProductIds } = this.state;
    const updated = toggleElInUniqueStringArray(selectedProductIds, productId);
    this.setState(prev => ({
      selectedProductIds: updated
    }));
  }

  hideShowProductDetail(selectedProductIds: ReadonlyArray<string>, product: GDAXProduct, key: number) {
    const isSelected = selectedProductIds.includes(product.id);
    console.log('PRODUCT ${product.id}, selected is: ', isSelected);
    if(isSelected) {
      return (
        <ProductDetail product={product} key={`${key}_prod`} />
      );
    }
    return null;
  }

  render() {
    const { qcProds } = this.props;
    const { selectedProductIds } = this.state;
    const buttonBaseStyle = {
      borderColor: 'black',
      borderLeftWidth: 5,
      borderRightWidth: 5,
    };

    return (
      <View key={`qc_view_${qcProds[0].quote_currency}`}> 
        {qcProds.map((qcp, i) => {
          const backgroundColor = selectedProductIds.includes(qcp.id) ? '#33E9FF' : 'white';
          const buttonStyle = Object.assign({}, buttonBaseStyle, {backgroundColor: backgroundColor});

          return (
            <View key={`prod_view_${i}`}>
              <Button
                key={i}
                title={qcp.quote_currency}
                color='black'
                onPress={this.toggleProductInfo.bind(null, qcp.id)}
                buttonStyle={buttonStyle}
              />
              {this.hideShowProductDetail(selectedProductIds, qcp, i)}
            </View>
          );
        })}
      </View>
    );
  }
}