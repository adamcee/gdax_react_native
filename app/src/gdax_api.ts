// gdax_api.ts
// interact with the gdax public gdax_api
import { getJSONResource } from './async';

export const GDAX_SANDBOX_HOST = 'https://api-public.sandbox.gdax.com';

export function getProducts() {
    return getJSONResource(GDAX_SANDBOX_HOST, 'products');
}