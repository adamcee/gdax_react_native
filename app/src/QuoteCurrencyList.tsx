// QuoteCurrencyList.tsx
import * as React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { GDAXProduct, GDAXProductMap } from './interfaces/gdax_product';
import { groupProductsByBaseCurrency } from './gdax_api_utils';
import { toggleElInUniqueStringArray } from './utils';

interface ComponentProps {
    qcProds: ReadonlyArray<GDAXProduct>;
}

interface ComponentState {
  products: ReadonlyArray<GDAXProduct>;
  selectedProductIds: Array<string>;
}

export default class QuoteCurrencyList extends React.Component<ComponentProps, ComponentState> {
  constructor(props: any) {
    super(props);
    this.state = {
      products: [],
      selectedProductIds: [],
    };

    this.toggleProductInfo = this.toggleProductInfo.bind(this);
  }

  // toggle if info for a product is displayed or not.
  toggleProductInfo(productId: string) {
    const { selectedProductIds } = this.state;
    const updated = toggleElInUniqueStringArray(selectedProductIds, productId);
    this.setState(prev => ({
      selectedProductIds: updated
    }));
  }

  // show detail info for a product.
  showProductInfo(product: GDAXProduct) {
    const productDetailNames = Object.keys(product);
    return (
      <View>
        {productDetailNames.map((detail, i) => 
          <Text key={i}>{detail}: {product[detail]}</Text>
        )}
      </View>
    );
  }

  render() {
    const { qcProds } = this.props;
    return (
      <View> 
        {qcProds.map((qcp, i) =>
          <Button
            key={i}
            title={qcp.quote_currency}
            onPress={this.toggleProductInfo.bind(null, qcp)}
          />
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
});