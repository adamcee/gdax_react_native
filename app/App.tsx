import * as React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { getProducts } from './src/gdax_api';
import { GDAXProduct, GDAXProductMap } from './src/interfaces/gdax_product';
import { groupProductsByBaseCurrency } from './src/gdax_api_utils';
import { toggleElInUniqueStringArray } from './src/utils';
import QuoteCurrencyList from './src/QuoteCurrencyList';

interface GDAXProductState {
  products: ReadonlyArray<GDAXProduct>;
  productsByBaseCurrency: GDAXProductMap;
  baseCurrencies: ReadonlyArray<string>;
}

interface SelectedProductsState {
  selectedBaseCurrencies: Array<string>;
}

interface AppState extends GDAXProductState, SelectedProductsState {
}

export default class App extends React.Component<{}, AppState> {
  constructor(props: any) {
    super(props);
    this.state = {
      products: [],
      productsByBaseCurrency: {},
      baseCurrencies: [],
      selectedBaseCurrencies: [],
    };

    this.toggleBaseCurrency = this.toggleBaseCurrency.bind(this);
  }

  componentDidMount() {
    console.log("ABOUT TO FETCH");
      const products = getProducts()
        .then(( json ) => {
          const groupedProducts: GDAXProductMap = groupProductsByBaseCurrency(json);
          const baseCurrencies: Array<string> = Object.keys(groupedProducts);
          this.setState({
            products: json,
            productsByBaseCurrency: groupedProducts,
            baseCurrencies: baseCurrencies,
          })
        })
        .catch(error => console.log(error));
  }

  // toggle if the quote currencies belonging to a given base currency are displayed or not.
  toggleBaseCurrency(baseCurrency: string) {
    console.log('STATE IS: ', this.state)
    const { selectedBaseCurrencies } = this.state;
    console.log('OLD ', selectedBaseCurrencies)
    const updated = toggleElInUniqueStringArray(selectedBaseCurrencies, baseCurrency);
    console.log('NEW ', updated)
    this.setState(prev => ({
      selectedBaseCurrencies: updated
    }));
  }

  // list all the base currencies
  listBaseCurrencies() {
    const { productsByBaseCurrency, baseCurrencies, selectedBaseCurrencies } = this.state;
    return baseCurrencies.map((bc, i) => 
      this.listBaseCurrency(bc, selectedBaseCurrencies, productsByBaseCurrency[bc])
    );
  }

  listBaseCurrency(bcName: string, selectedBcs: ReadonlyArray<string>, selectedBcProducts:  ReadonlyArray<GDAXProduct>) {
      const isSelected = selectedBcs.includes(bcName);
      const style = isSelected ? styles.selectedButton : styles.button
      return (
        <View>
          <Button
            key={bcName}
            title={bcName}
            onPress={this.toggleBaseCurrency.bind(null, bcName)}
          />
          {isSelected ? <QuoteCurrencyList qcProds={selectedBcProducts} /> : null }       
        </View>
      );
  }

  render() {
    const { products, productsByBaseCurrency, baseCurrencies } = this.state;
    return (
      <View style={styles.container}>
        <Text>Open up App.ts to start working on your app!</Text>
        <Text>Changes you make will automatically reload.</Text>
        <Text>Shake your phone to open the developer menu.</Text>
        <View> 
          {this.listBaseCurrencies()}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    backgroundColor: 'blue',
  },
  selectedButton: {
    backgroundColor: 'green',
  },
});