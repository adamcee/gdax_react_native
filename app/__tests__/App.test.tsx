import * as React from 'react';
import App from '../App';
import QuoteCurrencyList from '../src/components/QuoteCurrencyList';
import ProductDetail from '../src/components/ProductDetail';
import { groupProductsByBaseCurrency } from '../src/gdax_api_utils';
const mockGDAXProducts = require('../mocks/gdax_products.json');

import * as renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

it('<App /> renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('test child components which require props', () => {
  const groupedProducts = groupProductsByBaseCurrency(mockGDAXProducts);
  const baseCurrencies = Object.keys(groupedProducts);
  // all quote currences for the first base currency.
  const qcProducts = groupedProducts[baseCurrencies[0]];

  const someProduct = qcProducts[0];

  it('<QuoteCurrencyList /> renders correctly', () => {
    const tree = renderer.create(<QuoteCurrencyList qcProds={qcProducts} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('<ProductDetail /> renders correctly', () => {
    const tree = renderer.create(<ProductDetail product={someProduct} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});


