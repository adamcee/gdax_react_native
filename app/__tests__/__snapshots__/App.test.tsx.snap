// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingBottom": 80,
      "paddingTop": 40,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          Object {
            "fontSize": 27.5,
          },
          undefined,
          undefined,
          undefined,
          Object {},
          undefined,
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          },
        ]
      }
    >
      Select a GDAX Base Currency.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          Object {
            "fontSize": 27.5,
          },
          undefined,
          undefined,
          undefined,
          Object {},
          undefined,
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          },
        ]
      }
    >
      Then select a Quote Currency for product info.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          Object {
            "fontSize": 27.5,
          },
          undefined,
          undefined,
          undefined,
          Object {},
          undefined,
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          },
        ]
      }
    >
      Scroll and toggle.
    </Text>
    <View />
  </View>
</RCTScrollView>
`;

exports[`test child components which require props <ProductDetail /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "marginLeft": 20,
          "marginRight": 20,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        base_currency
      </Text>
      : 
      BTC
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        base_max_size
      </Text>
      : 
      10000
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        base_min_size
      </Text>
      : 
      0.01
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        cancel_only
      </Text>
      : 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        display_name
      </Text>
      : 
      BTC/USD
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        id
      </Text>
      : 
      BTC-USD
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        limit_only
      </Text>
      : 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        margin_enabled
      </Text>
      : 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        max_market_funds
      </Text>
      : 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        min_market_funds
      </Text>
      : 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        post_only
      </Text>
      : 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        quote_currency
      </Text>
      : 
      USD
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        quote_increment
      </Text>
      : 
      0.01
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        status
      </Text>
      : 
      online
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            undefined,
          ]
        }
      >
        status_message
      </Text>
      : 
    </Text>
  </View>
</View>
`;

exports[`test child components which require props <QuoteCurrencyList /> renders correctly 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "white",
                "borderColor": "black",
                "borderLeftWidth": 5,
                "borderRightWidth": 5,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "black",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "white",
                "borderColor": "black",
                "borderLeftWidth": 5,
                "borderRightWidth": 5,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "black",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            GBP
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "white",
                "borderColor": "black",
                "borderLeftWidth": 5,
                "borderRightWidth": 5,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "black",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            EUR
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
